cardano-tx-generator/src/Cardano/Benchmarking/Script/Core.hs

```diff
@@ -0,0 +1,214 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Benchmarking.Script.Core
+where
+
+import           Prelude
+import           Control.Monad
+import           Control.Monad.Trans.Except
+import           Control.Monad.IO.Class
+import           Control.Monad.Trans.Class
+import           Control.Concurrent (threadDelay)
+
+import           Cardano.Api ( AddressInEra, CardanoEra(..), InAnyCardanoEra(..), IsShelleyBasedEra, Tx
+                             , NetworkId(..), cardanoEra)
+
+import           Cardano.Benchmarking.GeneratorTx as Core
+                   (readSigningKey, runBenchmark, secureGenesisFund, splitFunds, txGenerator, TxGenError)
+import           Cardano.Benchmarking.Types as Core (NumberOfTxs(..), InitCooldown(..), SubmissionErrorPolicy(..))
+import           Cardano.Benchmarking.GeneratorTx.Tx as Core (keyAddress, Fund)
+import           Cardano.Benchmarking.GeneratorTx.LocalProtocolDefinition as Core (startProtocol)
+import           Cardano.Benchmarking.GeneratorTx.NodeToNode (ConnectClient, benchmarkConnectTxSubmit)
+import           Cardano.Benchmarking.OuroborosImports as Core
+                   (LocalSubmitTx, SigningKeyFile
+                   , getGenesis, protocolToNetworkId, protocolToCodecConfig, makeLocalConnectInfo, submitTxToNodeLocal)
+import           Cardano.Benchmarking.Tracer as Core (createTracers, btTxSubmit_, btN2N_, btConnect_, btSubmission_)
+
+import           Cardano.Benchmarking.Script.Env
+import           Cardano.Benchmarking.Script.Setters
+import           Cardano.Benchmarking.Script.Store
+
+startProtocol :: FilePath -> ActionM ()
+startProtocol filePath = do
+  (liftIO $ runExceptT $ Core.startProtocol filePath) >>= \case
+    Left err -> lift $ throwE $ CliError err     
+    Right (loggingLayer, protocol) -> do
+      set LoggingLayer loggingLayer
+      set Protocol protocol
+      set BenchTracers $ Core.createTracers loggingLayer
+      set Genesis $ Core.getGenesis protocol
+      set NetworkId $ protocolToNetworkId protocol
+
+readSigningKey :: KeyName -> SigningKeyFile -> ActionM ()
+readSigningKey name filePath =
+  (liftIO $ runExceptT $ Core.readSigningKey filePath) >>= \case
+    Left err -> liftTxGenError err
+    Right key -> setName name key
+
+keyAddress :: forall era. IsShelleyBasedEra era => AddressName -> KeyName -> CardanoEra era -> ActionM ()
+keyAddress addrName keyName era = do
+  key <- getName keyName
+  networkId <- get NetworkId
+  let addr = Core.keyAddress @ era networkId key
+  setName addrName (InAnyCardanoEra era addr)
+
+getLocalSubmitTx :: ActionM LocalSubmitTx
+getLocalSubmitTx = do
+  networkId <- get NetworkId
+  socket    <- getUser TLocalSocket
+  return $ submitTxToNodeLocal $ makeLocalConnectInfo networkId socket
+
+secureGenesisFund ::
+      FundName
+   -> AddressName
+   -> KeyName
+   -> ActionM ()
+secureGenesisFund fundName fundAddr genesisKey = do
```

Comment:
I think it was "Funds secured", so it probably should be `secureGenesisFunds`, if we want to keep this little piece of trivia : -)